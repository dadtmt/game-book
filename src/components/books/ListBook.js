import React from 'react'
import SelectBook from './SelectBook'

import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

import withLoading from '../../hoc/withLoading'

export const ListBook = ({ data }) => (
  <ul>
    {data.User.books.map(book => (
      <li key={book.id}>
        <SelectBook {...book} />
      </li>
    ))}
  </ul>
)

const USER_BOOKS = gql`
  query getUserBooks($id: ID!) {
    User(id: $id) {
      books(orderBy: createdAt_DESC) {
        id
        title
        imageUrl
        description
      }
    }
  }
`

export default graphql(USER_BOOKS, {
  options: ({ loggedInUser }) => ({ variables: { id: loggedInUser.id } })
})(withLoading(ListBook))
