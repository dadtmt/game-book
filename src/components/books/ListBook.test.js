import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

import { ListBook } from './ListBook'

describe('ListBook component', () => {
  it('component renders loading correctly', () => {
    const props = {
      data: {
        User: {
          books: [
            {
              id: 'ID_1',
              description: `Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam deditšnihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.`,
              imageUrl: 'http://lorempixel.com/400/200/',
              title: 'Book Title'
            },
            {
              id: 'ID_2',
              description: `Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam deditšnihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.`,
              title: 'Second Book Title'
            }
          ]
        }
      }
    }
    const tree = renderer
      .create(
        <BrowserRouter>
          <ListBook {...props} />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
