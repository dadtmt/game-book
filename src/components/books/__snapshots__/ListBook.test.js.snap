// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBook component component renders loading correctly 1`] = `
<ul>
  <li>
    <div>
      <a
        href="book/ID_1"
        onClick={[Function]}
      >
        Select this book link
      </a>
      <h2>
        Book Title
      </h2>
      <div
        style={
          Object {
            "backgroundImage": "url(http://lorempixel.com/400/200/)",
            "backgroundSize": "cover",
            "paddingBottom": "100%",
          }
        }
      />
      <div>
        Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam deditšnihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.
         
      </div>
    </div>
  </li>
  <li>
    <div>
      <a
        href="book/ID_2"
        onClick={[Function]}
      >
        Select this book link
      </a>
      <h2>
        Second Book Title
      </h2>
      <div>
        Verum ad istam omnem orationem brevis est defensio. Nam quoad aetas M. Caeli dare potuit isti suspicioni locum, fuit primum ipsius pudore, deinde etiam patris diligentia disciplinaque munita. Qui ut huic virilem togam deditšnihil dicam hoc loco de me; tantum sit, quantum vos existimatis; hoc dicam, hunc a patre continuo ad me esse deductum; nemo hunc M. Caelium in illo aetatis flore vidit nisi aut cum patre aut mecum aut in M. Crassi castissima domo, cum artibus honestissimis erudiretur.
         
      </div>
    </div>
  </li>
</ul>
`;
